ion-schema/src/nfa.rs

```diff
@@ -156,8 +155,11 @@ impl NfaEvaluation {
     }
 
     /// Validates provided ordered elements against referenced [Nfa]
-    pub fn validate_ordered_elements(&mut self, elements: &[Element], type_store: &TypeStore) {
-        let mut elements_iter = elements.iter().peekable();
+    pub fn validate_ordered_elements<'a, T: Iterator<Item = &'a Element>>(
+        &mut self,
+        mut elements_iter: Peekable<T>,
```

Comment:
TIL that `Peekable`'s impl of `peekable()` is not specialized to avoid producing a `Peekable<Peekable<T>>`.